### PR TITLE
small fix for log endpoint issue

### PR DESCRIPTION
**Pull request by rylandg**
Users will now get a relevant error when a bad log endpoint is provided.

### CODE REVIEW

**Comment by rylandg** [comment]
https://github.com/binaris/binaris/issues/384

File: test/spec.yml [hunk 1]
```diff
@@ -607,6 +607,9 @@
         err: |-
              Invalid offset format, unknown unit "daysf"
         exit: 1
+    -   in: BINARIS_LOG_ENDPOINT=fake.binaris.com bn logs amillionamillion
```
[the remark this answers is not in the document]
Comment (vogre) [reply test/spec.yml]: Fake domains tend to be problematic when ISP / Hotel inject their page on NXDOMAIN - so won't happen in CI, but will annoy someone one day. Hopefully they won't break `.invalid`.

[the remark this answers is not in the document]
Comment (arielshaqed) [reply test/spec.yml]: Ah, but the beauty of `invalid.` is that the hotel never sees it!  No query is sent (assuming a conforming resolver on the host):
>    3.  Name resolution APIs and libraries SHOULD recognize "invalid" names as special and SHOULD always return immediate negative        responses.  Name resolution APIs SHOULD NOT send queries for        "invalid" names to their configured caching DNS server(s).

[the remark this answers is not in the document]
Comment (vogre) [reply test/spec.yml]: `$ curl some-domain.invalid`  does send the query to `8.8.8.8` (tested with wireshark).

[the remark this answers is not in the document]
Comment (vogre) [reply test/spec.yml]: SHOULD aka. "RECOMMENDED"

**Comment by michaeladda** [comment]
@rylandg and progress on this?

**Comment by rylandg** [comment]
@michaeladda @arielshaqed fixed.

**Comment by michaeladda** [comment]
@rylandg this fails linting. did you even look at the CI error?

File: sdk/logs.js [hunk 2]
```diff
@@ -28,22 +35,23 @@ const logs = async function logs(functionName, apiKey, follow, startAfter, token
...
-      await msleep(backoff);
-    }
+    });
+
+    if (ok) return ok;
+    const backoffSec = backoffSecs.shift();
+    if (backoffSec === undefined) throw notOk;
+    msleep(backoffSec * 1000);
   }
 };
```
Comment (michaeladda) [inline sdk/logs.js]: maybe add unreachable comment?

**Pull request merged**